### PR TITLE
run e2e/qa/upgrade test on e2e queue

### PR DESCRIPTION
**Pull request by davejrt**
migrate tests off of vagrant and run them directly on a gcp vm.

Follow up is to remove any old references to these tests once we're confident they're suitable and not buggy.

Depends on sourcegraph/infrastructure#2945

### CODE REVIEW

File: enterprise/dev/ci/internal/ci/operations.go [hunk 1]
```diff
@@ -411,7 +411,7 @@ const vagrantServiceAccount = "buildkite@sourcegraph-ci.iam.gserviceaccount.com"
 func serverE2E(candidateTag string) operations.Operation {
 	return func(p *bk.Pipeline) {
 		p.AddStep(":chromium: Sourcegraph E2E",
-			bk.Agent("queue", "baremetal"),
+			bk.Agent("queue", "e2e-qa"),
```
Comment (bobheadxi) [inline enterprise/dev/ci/internal/ci/operations.go:414]: Could we name this queue something more general than e2e-qa? For example, I expect a potential outcome of the E2E test updates is that E2E and QA pipelines will eventually be merged - we should have a naming that will survive such a change and still make sense

File: enterprise/dev/ci/internal/ci/operations.go [hunk 2]
```diff
@@ -427,15 +427,15 @@ func serverE2E(candidateTag string) operations.Operation {
 			bk.Env("TEST_USER_EMAIL", "test@sourcegraph.com"),
 			bk.Env("TEST_USER_PASSWORD", "supersecurepassword"),
 			bk.Env("INCLUDE_ADMIN_ONBOARDING", "false"),
-			bk.Cmd(".buildkite/vagrant-run.sh sourcegraph-e2e"),
+			bk.Cmd("./dev/ci/test/e2e/test.sh"),
```
Comment (bobheadxi) [inline enterprise/dev/ci/internal/ci/operations.go:430]: Nice 😁

**Comment by bobheadxi** [comment]
I've created a retroactive issue tracking work on going vagrant-free, since I noticed there isn't one yet: https://github.com/sourcegraph/sourcegraph/issues/28709

File: enterprise/dev/ci/internal/ci/operations.go (hunk 2 above)
Comment (jhchabran) [reply enterprise/dev/ci/internal/ci/operations.go:430]: This is gonna be good 😄

File: enterprise/dev/ci/internal/ci/operations.go (hunk 1 above)
Comment (jhchabran) [reply enterprise/dev/ci/internal/ci/operations.go:414]: Good point, this are always painful to move arounds and errors are messing up with the overall stability. I think "baremetal" was more meaningful as it indicated which class of agents are running these jobs, not making any distinctions about what they actually do. WDYT?

File: dev/ci/test/cleanup-display.sh [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
 
-PID=$(pgrep ffmpeg)
-kill "$PID"
+procs=(ffmpeg Xvfb)
+
+for p in "${procs[@]}"; do pgrep "$p" | xargs kill; done
```
Comment (jhchabran) [inline dev/ci/test/cleanup-display.sh:5]: Just being curious, what motivates the change here?

Comment (davejrt) [reply dev/ci/test/cleanup-display.sh:5]: The `Xvfb` process was leaving a display running and messing with a test being run again. These still aren't "stateless" so this is just ensuring we clean up as much as possible.

File: enterprise/dev/ci/internal/ci/operations.go (hunk 1 above)
Comment (davejrt) [reply enterprise/dev/ci/internal/ci/operations.go:414]: In that case, I propose that these be called baremetal, and queue running the remaining [docker tests](https://k8s.sgdev.org/github.com/sourcegraph/deploy-sourcegraph-docker/-/blob/.buildkite/pipeline.yaml?L8) be changed to `vagrant` in order to correctly distinguish between the two for now.

File: dev/ci/test/cleanup-display.sh (hunk 3 above)
Comment (jhchabran) [reply dev/ci/test/cleanup-display.sh:5]: Oh my bad, I read `PID=$(pgrep ffmpeg)` as `PID=$(Xvfb ffmpeg)` and missed the quotes on the `kill` call.

**Pull request merged**